### ISSUE TITLE
BO - Products page - Specific prices - MySQL 8 - The Combination applied to specific price is NOK

### ISSUE THREAD

**Issue by khouloudbelguith** [PrestaShop/PrestaShop#23973]
#### Describe the bug

In the BO > Products page, when we try to add a specific price to a combination => the combination selected is NOK
Thanks @nesrineabdmouleh 

#### Steps to Reproduce

1. Go to BO > Catalog >Products page > Edit a product with combinations
2. In the Pricing tab > Add a specific price 
3. Select a combination 
4. Set an Impact on price
5. Click on Apply
6. See error: the combination is not correct, it is the next one selected

**Screenshots**

https://drive.google.com/file/d/1k3BBsf1zrgRXhAr6QCZLPlkHj1VJEHW_/view?usp=sharing

#### Additional information

* PrestaShop version: develop
* PHP version: 7.2
* MySQL 8 (according to below feedback, MySQL 5 is fine)

**Comment by isaacrj** [comment]
We manage a few PS installations and we have had this issue on some of them for months. See #20949 
We have noticed that the installations that fail are those where the query
```
SELECT * FROM `ps_product_attribute` WHERE id_product = <id_product>
```
returns its results NOT ordered by its primary key (id_product_attribute)

Just my 2 cents, in case that there is a query somewhere missing an ORDER BY id_product_attribute.

**Comment by TBurtin** [comment]
We have the issue on a site, using MySQL 8.0 version.

With the same site with MySQL 5.7, no issue.

Maybe a difference with order by default on Mysql 8 ?

**Comment by matks** [comment]
@khouloudbelguith Can you confirm what MySQL version you are using?

**Comment by khouloudbelguith** [comment]
Hi @matks ,

My MySQL version is  8.0.22
![image](https://user-images.githubusercontent.com/16067358/116892444-212df600-ac30-11eb-8992-01d5ed20473b.png)

Thank you!

**Comment by matks** [comment]
@khouloudbelguith Thank you :) so we are sure this is related to MySQL 8 as suggested by @TBurtin

**Comment by Progi1984** [comment]
Fixed by @daresh in #24089
